traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/cache/Cache.java

```diff
@@ -228,9 +234,53 @@ public InetAddress getIp6() {
 		return ip6;
 	}
 
+	@SuppressWarnings({"PMD.CyclomaticComplexity", "PMD.NPathComplexity"})
 	public void setState(final JsonNode state) {
 		final boolean isAvailable = JsonUtils.optBoolean(state, "isAvailable", true);
+		final boolean ipv4Available = JsonUtils.optBoolean(state, "ipv4Available", true);
+		final boolean ipv6Available = JsonUtils.optBoolean(state, "ipv6Available", true);
+
+		final List<InetRecord> newlyAvailable = new ArrayList<>();
+		final List<InetRecord> newlyUnavailable = new ArrayList<>();
+
+		if (ipv4Available && !ipv6Available) {
+			this.ipAvailableVersions = IpVersions.IPV4ONLY;
+		} else if (ipv6Available && !ipv4Available) {
+			this.ipAvailableVersions = IpVersions.IPV6ONLY;
+		} else {
+			this.ipAvailableVersions = IpVersions.BOTH;
+		}
+
+		for (final InetRecord record : ipAddresses) {
+			if (record.getAddress() == ip4 && !ipv4Available) {
+				newlyUnavailable.add(record);
+				LOGGER.info("disabling IPv4 address " + record.getAddress());
```

Comment:
`INFO` seems like the wrong logging level for this kind of information. It might make more sense if it was just logging the state changes (e.g. disabled -> enabled and vice versa), but logging states for every server for every state poll from TM seems like too much. Should we make this `DEBUG` instead?

Reply:
this should only log the state changes. the top loop logs if a previously enabled ip address is now disabled, and the bottom does the opposite. this might still be too much for an INFO though. im good with changing it to DEBUG

Reply:
done